src/Sparrow/Json/LazyStringValue.cs

```diff
@@ -903,5 +903,15 @@ public void Renew(string str, byte* buffer, int size)
             IsDisposed = false;
             AllocatedMemoryData = null;
         }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public bool IsControlCodeCharacter(out byte b)
+        {
+            Debug.Assert(Size == 1);
+
+            b = Buffer[0];
+            // control code characters
+            return b < 32 || (b >= 127 && b <= 159);
```

Comment:
Why is a single line string special in this way? What happens if we have a space and then a control char?

Reply:
@ayende Starting to look into this it seems that this is not the right place to handle this.
We should construct the LazyStringValue with those control character already escaped meaning we will need to go over all the places that construct the escaped character and make sure we escape control characters, right?

Reply:
That should be the case when we build LZV, yes.

Reply:
I think that we should not treat that on server side.
We did not validate incoming jsons and that train has already left the platform, due to backward compatability issues (imagine refusing requests from older clients by newer servers etc.)
We do need to make sure that the clients will send a valid json from now on